### DISCUSSION TITLE
Can a sensor materialize assets without specifying a job to do it?

### DISCUSSION THREAD

**Discussion by garethbrickman** [dagster-io/dagster#23515] [Q&A] [answered]
I have several assets that I'd like to be materialized by a sensor and I'll have logic in my code to decide which to materialize at the time the sensor evaluates. Is this possible as I can't specify jobs in advance to do this?

**Comment by garethbrickman** [comment, marked as answer]
Yes, you can use the `asset_selection` parameter in the ­[@sensor decorator](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.sensor) and the [RunRequest object ](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.RunRequest)to bypass needing to specify a job.

Here's an example where all assets are defined in the decorator and then code logic in the sensor decides which specific assets to materialize in the RunRequests:

```python
@sensor(asset_selection=AssetSelection.all())
def my_sensor(context):
    # logic to decide which assets get materialized by the Run Request
    yield RunRequest(asset_selection=[asset_a.key, asset_b.key])
```

Here's an example where a specific list of assets is defined in the decorator and then code logic in the sensor decides which specific assets to materialize in the RunRequests:

```python
@sensor(asset_selection=AssetSelection.assets("asset_a", "asset_b", "asset_c"))
def my_sensor(context):
    # logic to decide which assets get materialized by the Run Request
    yield RunRequest(asset_selection=[asset_c.key])
```

Here's a contrived example using Python's `random` library showing how asset selection for materialization can be dynamic:

```python
@sensor(asset_selection=AssetSelection.assets("asset_a", "asset_b"))
def dynamic_asset_sensor(context):
    sensor_evaluation_result = random.choice([True, False])

    if sensor_evaluation_result:
        yield RunRequest(
            asset_selection=[asset_a.key]
        )
    else:
       yield RunRequest(
            asset_selection=[asset_b.key]
        )
```

See [examples here](https://docs.dagster.io/_apidocs/assets#dagster.AssetSelection) for the kinds of asset selections that can be done using the `AssetSelection` class.

**Comment by Terroface** [comment]
@garethbrickman what about when using `@run_status_sensor()`? It has no `asset_selection` argument.
